src/components/date/__internal__/date-formats/index.js

```diff
@@ -47,24 +70,61 @@ const generateFormats = (formatArray) =>
     return array;
   }, []);
 
+const getOutputFormatForLocale = (localeCode) => {
+  const formatMap = {
+    day: "dd",
+    month: "MM",
+    year: "yyyy",
+  };
+
+  const formatter = new Intl.DateTimeFormat(localeCode);
+
+  const format = formatter
+    .formatToParts(new Date())
+    .map(({ type, value }) => {
+      if (type !== "literal") return formatMap[type];
+
+      return value;
+    })
+    .join("");
+
+  if (localeCode === "bg-BG") {
+    // this locale adds an additional char that has no effect on the output formatting
```

Comment:
Are there any other locales we support that add additional characters that we would need to cover?